ADR/propagate-pull-secrets.md

```diff
@@ -0,0 +1,54 @@
+# 0. Support deploying images from Private Quay.io repositories
+
+Date: 2023-03-16
+
+## Status
+
+Provisional
+
+## Context
+
+* As a user, I would expect StoneSoup to generate a private Quay.io image repository if my source code repository is  private.
+* As a user, I would expect StoneSoup to gracefully deploy images that were pushed to private Quay.io repositories in every relevant environment.
+
+
+## Decision
+
+### Overview
+
+1. This proposal builds on the capabilties of the [image-controller](https://redhat-appstudio.github.io/book/book/image-controller.html) which sets up a Quay.io image repository per `Component` and 'downloads' a robot account token into the user's namespace for builds to be able to push to it.
+
+2. This proposal builds on the capabilities of SPI's `RemoteSecret` API. See https://github.com/redhat-appstudio/book/pull/70/files
+
+
+#### When a new `Component` is created
+
+1. image-controller creates a new Quay.io image repository
+2. image-controller provisions a new robot account token that would only be able to "pull" the `Component`'s image.
```

Comment:
Not sure what does robot accout token provisioning mean. I thought it can be only generated on quay.io web page and copy-pasted...Or there is some more automated way? 
Anyway, we can leverage storing this secret data on secret storage using RemoteSecret on this stage by for example:
- create upload secret with the robot accout token and `Application`/`Component` annotations. 
- It's controller will upload token data to the storage and create RemoteSecret with corresponding `Application`/`Component` labels (upload secret will be immediately deleted).

Reply:
👍